packages/core/database/lib/query/helpers/streams/readable.js

```diff
@@ -0,0 +1,175 @@
+'use strict';
+
+const { Readable } = require('stream');
+
+const kQueryDone = Symbol('kOpDone');
+const kPerformingQuery = Symbol('kPerformingQuery');
+
+const { applyPopulate } = require('../populate');
+const { fromRow } = require('../transform');
+
+class ReadableStrapiQuery extends Readable {
+  /**
+   * @param {object} options
+   * @param {ReturnType<typeof import('../../query-builder')>} options.qb The strapi query builder instance
+   * @param {string} options.uid The model uid
+   * @param {import('../../../index').Database} options.db The Database instance
+   * @param {boolean} [options.mapResults] The maximum number of entities to fetch per query
+   * @param {number} [options.batchSize] The maximum number of entities to fetch per query
+   */
+  constructor({ qb, db, uid, mapResults = true, batchSize = 500 }) {
+    super({ objectMode: true, highWaterMark: batchSize });
+
+    // Extract offset & limit from the query-builder's state
+    const { offset, limit } = qb.state;
+
+    // Original offset value
+    this._offset = offset === undefined ? 0 : offset;
+
+    // Max amount of entities to fetch, force null as undefined value
+    this._limit = offset === undefined ? null : limit;
```

Comment:
Should this be checking `limit` instead of `offset`? (Also, same comment as above with a more robust `isFinite` check)

Reply:
It should be checking limit indeed :man_facepalming:

Reply:
Fixed